src/convertToObject.js

```diff
@@ -3,16 +3,29 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles (see an example in
+ * [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const splittedString = sourceString.split(';');
+  const stylesObj = {};
+
+  const propsNamesAndValues = splittedString
+    .map(prop => prop.trim().split(':'))
+    .filter(prop => prop[0]);
+
+  propsNamesAndValues.forEach(prop => {
```

Comment:
what is the goal of interruption of chaining? It is possible to use `forEach` after `filter` here

Reply:
Thought it'd be less confusing